шабалин home work

Andreyshabalinn opened the pull request.

Andreyshabalinn committed:
Add files via upload

Andreyshabalinn committed:
Добавил SQL скрипт по созданию бд, таблиц и тд))

sunzo-digital commented on Dz.sql:
```diff
@@ -0,0 +1,156 @@
+);
+
+CREATE TABLE PositiONs(
+id INT PRIMARY KEY auto_increment not null,
```
похоже, что из-за того, что здесь нет отступов, возникает ошибка:
`ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'CREATE TABLE positiON_orders(                                   order_id int,                                                                                                                                                                                                      positiON_id int,                                                                                                                                                                                                   foreign key (ord' at line 15`

убедись, что sql файл отрабатывает корректно

sunzo-digital commented on Dz.sql:
```diff
@@ -0,0 +1,156 @@
+)
+
+ /*Спасибо обсуждению в телеге за эту таблицу :) */
+CREATE TABLE positiON_orders(
```
здесь синтаксическая ошибка:
`ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'CREATE TABLE positiON_orders(                                   order_id int,                                                                                                                                                                                                      positiON_id int,                                                                                                                                                                                                   foreign key (ord' at line 15...`

sunzo-digital reviewed (commented):
синтаксическая ошибка

Andreyshabalinn committed:
Пофиксил ошибки

Там было много пропущенных ;
Я просто создавал пошагово (Через выделение и выполнение запроса), поэтому не обратил внимание:(

sunzo-digital reviewed (commented):
Похоже, что теперь в некоторых места имена таблиц написаны с ошибкой:
```
Query OK, 1 row affected (0.02 sec)

Database changed
Query OK, 0 rows affected (0.05 sec)

Query OK, 0 rows affected (0.05 sec)

Query OK, 0 rows affected (0.07 sec)

Query OK, 0 rows affected (0.06 sec)

Query OK, 0 rows affected (0.09 sec)

Query OK, 3 rows affected (0.01 sec)
Records: 3  Duplicates: 0  Warnings: 0

Query OK, 9 rows affected (0.01 sec)
Records: 9  Duplicates: 0  Warnings: 0

ERROR 1146 (42S02): Table 'food_delivery.clients' doesn't exist
ERROR 1146 (42S02): Table 'food_delivery.orders' doesn't exist
ERROR 1146 (42S02): Table 'food_delivery.positions' doesn't exist
ERROR 1146 (42S02): Table 'food_delivery.clients' doesn't exist
ERROR 1146 (42S02): Table 'food_delivery.clients' doesn't exist
Query OK, 1 row affected (0.00 sec)

Query OK, 1 row affected (0.02 sec)

ERROR 1146 (42S02): Table 'food_delivery.positions' doesn't exist
ERROR 1146 (42S02): Table 'food_delivery.orders' doesn't exist
```
вот так выглядят твои таблицы после применения файла
```
mysql> show tables;
+-------------------------+
| Tables_in_food_delivery |
+-------------------------+
| Clients                 |
| Orders                  |
| Partners                |
| Positions               |
| position_orders         |
+-------------------------+
```

Andreyshabalinn committed:
Пофиксил названия таблиц

Подозреваю проблема была в заглавных/строчных буквах, поменял
Затрудняюсь назвать истинную причину, потому что у меня до фикса работало

sunzo-digital reviewed (commented):
```
Query OK, 1 row affected (0.02 sec)

Database changed
Query OK, 0 rows affected (0.07 sec)

Query OK, 0 rows affected (0.07 sec)

Query OK, 0 rows affected (0.06 sec)

Query OK, 0 rows affected (0.10 sec)

Query OK, 0 rows affected (0.07 sec)

Query OK, 3 rows affected (0.01 sec)
Records: 3  Duplicates: 0  Warnings: 0

Query OK, 9 rows affected (0.01 sec)
Records: 9  Duplicates: 0  Warnings: 0

Query OK, 3 rows affected (0.01 sec)
Records: 3  Duplicates: 0  Warnings: 0

Query OK, 6 rows affected (0.01 sec)
Records: 6  Duplicates: 0  Warnings: 0

Query OK, 13 rows affected (0.01 sec)
Records: 13  Duplicates: 0  Warnings: 0

ERROR 1146 (42S02): Table 'food_delivery.partners' doesn't exist
ERROR 1146 (42S02): Table 'food_delivery.partners' doesn't exist
Query OK, 1 row affected (0.00 sec)

Query OK, 1 row affected (0.01 sec)

ERROR 1054 (42S22): Unknown column 'partners.id' in 'IN/ALL/ANY subquery'
Empty set (0.00 sec)
```

Таблицы твоей БД сейчас
```
+-------------------------+
| Tables_in_food_delivery |
+-------------------------+
| Clients                 |
| Orders                  |
| Partners                |
| Positions               |
| position_orders         |
+-------------------------+
```

Andreyshabalinn committed:
Поправил названия таблиц
